src/UnigramWordPredictor.java

```diff
@@ -99,11 +114,26 @@ public void train(Scanner scanner) {
    * @return the predicted next word, or null if no prediction can be made
    */
   public String predictNextWord(List<String> context) {
-    // TODO: Return a predicted word given the words preceding it
-    // Hint: only the last word in context should be looked at
-    return null;
+    Random random = new Random();
+    String nextWord = "";
+
+    // Grab the last word generated in the context
+    String lastWord = context.get(context.size()-1);
+
+    // Checks to see if the word has a key linked to it in neighbor map
+    // returns null if false
+    // returns a randomly selected word from the list mapped to each word
+    if(!neighborMap.containsKey(lastWord)){
+      return null;
+    }else{
+      List<String> contextList = neighborMap.get(lastWord);
+      int nextWordIndex = random.nextInt(contextList.size());
+ 
+      nextWord = contextList.get(nextWordIndex);
+    }
```

Comment:
You can actually omit `else` here. The function will return if the `if` statement is triggered, so we know that it must contain the key if we reach line 129.